### ISSUE TITLE
Hose spray on steel beam causes instability

### ISSUE THREAD

**Issue by mcgratta** [firemodels/fds#8558]
[CoolingTest5.txt](https://github.com/firemodels/fds/files/4886903/CoolingTest5.txt)
This case becomes unstable within about 5 min of starting. It is a hose spray on a steel beam. No heat sources, but the beam temperature oscillates wildly. Run the case on 8 MPI processes and you will see the problem within a few minutes. The job eventually fails with a WARNING about droplet temperatures and a seg fault, but this is a red herring. The trouble starts long before the droplets act up because the gas temperature oscillates wildly.

Simo, I included you because you might have some insight on what happens when a lot of water splashes onto a cold surface. I suspect it's a numerical instability in the wall temperature solver. Like a really  large source or convection term. I don't think the problem is related to anything done recently because the reported error was based on the last release of FDS.

**Comment by shostikk** [comment]
Kevin, we discussed this case this morning with Rahul. This is part of our project. 😁

The case has artificially high heat transfer coefficient for CPUA, optimized for reproducing experiment, but possibly unphysical.



Sent from my Huawei phone


-------- Original Message --------
Subject: Re: [firemodels/fds] Hose spray on steel beam causes instability (#8558)
From: Kevin McGrattan
To: firemodels/fds
CC: Hostikka Simo ,Assign


Assigned #8558<https://github.com/firemodels/fds/issues/8558> to @shostikk<https://github.com/shostikk>.

**Comment by mcgratta** [comment]
I thought the name was familiar. There should be a natural upper bound in CPUA. I'll take a look at this.

**Comment by shostikk** [comment]
It turned out to be difficult to match the experimental cooling rate. We may end up compensating the lack of 3D conduction and some real water behaviours by turning up H_SOLID_TO_DROPLET at PART line. The upper bound should be derived from numerical stability requirement, if possible.

**Comment by rmcdermo** [comment]
CHECK_HT is not applied to particles and the same stability requirement should apply.  I really do not want to see four different stability requirements/cutoffs in the code.  We already have:
1. Divergence
2. CHECK_HT
3. Something is done for droplets to AIR that something slightly different than the current CHECK_HT (line 2875 of part.f90)
4. Potentially new droplet to solid (this issue).

Let's tighten this up.

**Comment by mcgratta** [comment]
I don't believe that this issue is about time steps. I believe it has to do with the large amount of water hitting the solid surface causing fictitious increases and decreases in droplet and wall temperature. I think this is some simple bug, but the droplet heating routine has become so complicated I cannot figure out where the problem lies.

**Comment by rmcdermo** [comment]
The temperature swings would be instabilities of an explicit code.  I agree that marching at a sufficiently small time step might not be practical.  But I want to be careful about how we overcome this issue.  We don't want arbitrary cutoffs.

**Comment by mcgratta** [comment]
I have traced a single droplet from the hose nozzle until it strikes the steel beam. It is about 3 mm in diameter, and its initial temperature is 5 C, as is the ambient air. By the time the droplet  reaches the beam, it is about 4.5 C. When the droplet hits the beam, its temperature jumps up to 65 C (not sure why, this seems counter-intuitive) and then via iteration the steel temperature decreases back down to about 10 C. By then the droplet temperature is still about 54 C, and still in this same time step the steel now drops to -730 C, and clearly an instability is occurring. Unless there is a simple bug here, I cannot think of how to fix this without a total overhaul. My first thought is to not try to update the wall temperature in the particle routine, but rather pass to the 1-D heat transfer routine the water temperature and heat transfer coefficient, and apply the convection from water just like we do the convection from air in the C-N scheme. But I have to think this through a bit.

**Comment by drjfloyd** [comment]
I haven't had a chance yet to delve into this.  I should have time today.

We don't update the wall temperature in the particle routine. We only estimate the change in temperature of the first wall cell in order to get the convection to/from the droplet.

**Comment by mcgratta** [comment]
OK, I'm tracing particles to see where the problem is. Right now I'm
looking hard at MINIMUM_FILM_THICKNESS. This is a SWAG from 25 years ago
that assumes how much a single particle will "splat". This can definitely
affect the heat transfer.

>

**Comment by mcgratta** [comment]
Ugh. The units on H_SOLID_TO_DROPLET are wrong. The User's Guide says the
default is 0.3 kJ/m2/K (which should be kW/m2/K). This number is multiplied
by 1000 and then used within the droplet routine. The input file for this
case has a value of 1800 instead of 1.8. That will cause great havoc.

I will fix all this stuff.

>

**Comment by drjfloyd** [comment]
An h of 1.8 MW/m^2/K is going to cause challenges, but I'll still look to make sure there aren't other issues,

**Comment by mcgratta** [comment]
No doubt. I think the MINIMUM_FILM_THICKNESS should be an input, and
default to 0.001, which is the same size we assume when a droplet hits a
wall.

Also, I think we should add a section in the Appendix of the Tech Guide to
describe the droplet heating and evap algorithm. What we have now is very
short. And whatever stuff that doesn't need to be in there should be taken
out. That routine is very long and hard to follow. If this problem weren't
so dumb, I could never have fixed it.

>

**Comment by mcgratta** [comment]
I'll let firebot test the changes. The only real change is that `MINIMUM_FILM_THICKNESS` is now 0.001 m instead of 0.00001 m. And the units of `HEAT_TRANSFER_COEFFICIENT_SOLID` on the `PART` line is W/m2/K, to be consistent with other input HTCs.

**Comment by drjfloyd** [comment]
I may have an error in setting up the matrix for a wall.

**Comment by mcgratta** [comment]
I restored `MINIMUM_FILM_THICKNESS` to its original value of 0.00001 m. This is meant to be a lower bound, not an actual estimate of the film thickness. For water, the film thickness can vary from microns to millimeters depending on the surface conditions. In cases where convective heat transfer is important, this value can be important because it determines the area of contact between liquid and solid.

I'll leave the case open until we tie up all loose ends and test with firebot. At the moment, nothing has changed other than name changes and units, but functionality should be the same.

**Comment by drjfloyd** [comment]
I did find a bug and fixing it improves the behavior of the original case somewhat; however, the original h of 1.8 MW/m^2/K still causes havoc. Not quite enough to kill the simulation quickly, but still enough to spike the steel temperature fairly high. I need to test the fixes against the water_evap cases.

I am also adding MASS_TRANSFER_COEFFICIENT and HEAT_TRANSFER_COEFFICIENT_GAS to PART to go along with HEAT_TRANSFER_COEFFICIENT_SOLID.  I think it will be useful to be able to control those.

**Comment by rkj15** [comment]
The heat transfer coefficient I specified in the input file is a mistake with the units, as I intended to have 1.8 kW/m^2/K. I don't think I will have to use a value of 1.8 MW/m^2/K for my application.

**Comment by mcgratta** [comment]
Jason, set MTC and HTC_GAS to -1 by default, which means we should apply a
model to calculate them. HTC_SOLID works this way too, but it is 300 by
default, meaning that we want the constant coefficient for the solid-gas
transfer.

>

**Comment by drjfloyd** [comment]
Pushed up the bug fix and the new keywords.

**Comment by mcgratta** [comment]
Are these new parameters in the User's Guide?

>

**Comment by drjfloyd** [comment]
yes

**Comment by mcgratta** [comment]
Are the units for h_m kg/m2/s or m/s? In the equation in the Tech Guide, it appears to be m/s.

**Comment by drjfloyd** [comment]
you are correct. I'll fix

**Comment by mcgratta** [comment]
Great, thanks.

**Comment by drjfloyd** [comment]
I updated the guide. I added an appendix chapter for evaporation and modified that Lagrangian chapter section on evaporation.  The appendix has a couple of large equations, what is a good way to force a line break? I tried // but that gave an error.

**Comment by mcgratta** [comment]
Use the latex environment called eqnarray.

\begin{eqnarray}
y &=& x + c  \\
z&=& u + 2
\end{eqnarray}

>

**Comment by drjfloyd** [comment]
As a test I did this (the actual equation has a lot of terms inside the "()" )
\begin{eqnarray}
x &=& y (1 \\
&-& 2)
\end{eqnarray}

I get an equation number for each line rather than one number for the entire equation

**Comment by drjfloyd** [comment]
I think I figured it out.  \nonumber on the first line

**Comment by drjfloyd** [comment]
Also need to add \right. and \left. to stop error messages due to splitting a [] over two lines.
